README.md

```diff
@@ -55,13 +57,165 @@ And in your modules, apply the `android-apt` plugin and add the remixer dependen
 }
 ```
 
 Notice the dependency on `remixer_annotation` is a `provided` clause instead of `compile`, this is on purpose as this is not a regular dependency but a compiler plugin.
 
+### Global remixer set up
+If you have not subclassed the application class it is recommended you do it since this is a one-time global initialization.
+
+In your application class you need to add RemixerActivityLifecycleCallbacks as an ActivityLifecycleCallbacks instance, so Remixer knows to remove old variables and triggers when activities are destroyed, to avoid leaks.
+```java
+class MyApplication extends android.app.Application {
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    registerActivityLifecycleCallbacks(RemixerActivityLifecycleCallbacks.getInstance());
+  }
+}
+```
+
+### How to use it in an activity
+
+__Only in the activities where you're using remixer__
+
+You need to add a few lines at the end of your `Activity.onCreate()`
+
+1. `RemixerBinder.bind(this);` creates, initializes and sets up all the Variables and trigger you define in this activity.
+2. `RemixerFragment remixerFragment = RemixerFragment.newInstance();` creates the fragment that will be shown when remixer is invoked, then you need at least one of the following:
+  - A variation of `remixerFragment.attachToGesture(this, Direction.UP, 3);`, this example ties showing the Remixer Fragment on a 3-finger swipe up.
+  - `remixerFragment.attachToButton(this, someButtonObject);`, this makes the OnClickListener for a button open the Remixer fragment.
+
+Your `Activity.onCreate` may look like this:
+```java
+@Override
+protected void onCreate(Bundle savedInstanceState) {
+  super.onCreate(savedInstanceState);
+  //...
+  remixerButton = (Button) findViewById(R.id.button);
+  RemixerBinder.bind(this);
+  RemixerFragment remixerFragment = RemixerFragment.newInstance();
+  remixerFragment.attachToGesture(this, Direction.UP, 3);
+  remixerFragment.attachToButton(this, remixerButton);
+}
+```
+
+#### Define variables
+
+In order to define variables you only need to write methods that take one argument of the correct type and annotate them. The methods contain your logic to handle changes to these variables (update the UI accordingly). You can rest assured those methods will run in the main UI thread.
+
+There are a few very simple examples here, but you should look at the [example](https://github.com/material-foundation/material-remixer-android/blob/develop/remixer_example/src/main/java/com/google/android/apps/remixer/MainActivity.java) [activities](https://github.com/material-foundation/material-remixer-android/blob/develop/remixer_example/src/main/java/com/google/android/apps/remixer/BoxActivity.java) and [documentation for these annotations](https://github.com/material-foundation/material-remixer-android/tree/develop/remixer_core/src/main/java/com/google/android/libraries/remixer/annotation) for more information.
```

Comment:
I wouldn't link [example] [activities] in that way for clarity reasons. Maybe just link both words to `https://github.com/material-foundation/material-remixer-android/blob/develop/remixer_example/src/main/java/com/google/android/apps/remixer`